### ISSUE TITLE
Support for NetStandard 2.0 and .Net 6

### ISSUE THREAD

**Issue by DrNeil** [rubicon-oss/CoreRhinoMocks#2]
It would be good if this project could now support NetStandard 2.0 
Then CoreRhinoMocks can be used in both NetStandard and .NET 6.0 projects
I have raised a PR here for this [here](https://github.com/rubicon-oss/CoreRhinoMocks/pull/1)

Thanks for doing the heavy lifting to get RhinoMocks to .Net core

**Comment by MichaelKetting** [comment]
Thanks for raising this issue.

Based on my experience, the project already supports .NET 6 since .NET 6 should be backwards compatible to .NET Core 3.1 as far as Rhino.Mocks goes. We used it with .NET 5.0 without any issues. Did you experience an issue?

Could you please expand on the usecase for .NET Standard in unit tests? I want to make sure this is actually a helpful usecase. Since the unit test projects always execute within a particular target runtime, I need to target .NET Framework or .NET Core / .NET 6 anyway, making .NET Standard much less helpful in unit test scenarios.

**Comment by DrNeil** [comment]
By targeting NetStandard 2.0 the library offers most coverage across the SDKs (old framework and new .NET 5 and 6) 
[Net Standard versions](https://dotnet.microsoft.com/en-us/platform/dotnet-standard#versions)
As .NET Core 3.1 will end support this year it makes sense to move to an SDK that will be supported for longer.
[.NET Support](https://dotnet.microsoft.com/platform/support/policy/dotnet-core?msclkid=177388c3cf7311ec9acb822e6c5b3f5f)

An alternative would be to move to .NET 6, however that might preclude some other people from getting benefit from this work.

**Comment by MichaelKetting** [comment]
There's some APIs that are .Net-Framework-only so we will need to keep the .NET 3.5.1 support, too. I don't mind adding .NET 6 as a dedicated target next to the other targets but I don't think removing the older targets would be a safe choice.

Regarding the most-common use for the library, using it in an environment that does not target .NET Framework or .NET Core 3.1 (or in the future, .NET 6 compatible) might be problematic. There have been no tests for those and with a library that relies on runtime code generation, beign too open is potentially problematic. 

So to summarize, adding .NET 6 support is fine by me, but we need to keep the rest as-is.

**Comment by DrNeil** [comment]
hi, changed the PR to now multitarget for net6 and netstandard - best of all worlds
